### PR TITLE
Upgrade AMQP library to 2.5.8

### PR DESCRIPTION
**Pull request by azabbasi**

### CODE REVIEW

File: iothub/service/src/ServiceClient.cs [hunk 1]
```diff
@@ -272,7 +272,9 @@ public virtual async Task OpenAsync()
         {
             Logging.Enter(this, $"Opening AmqpServiceClient", nameof(OpenAsync));
 
-            await _faultTolerantSendingLink.OpenAsync(_openTimeout).ConfigureAwait(false);
+            using var ctx = new CancellationTokenSource(_openTimeout);
+
+            await _faultTolerantSendingLink.OpenAsync(ctx.Token).ConfigureAwait(false);
```
Comment (azabbasi) [inline iothub/service/src/ServiceClient.cs:277]: This method no longer supports timespans.

**Review by drwill-ms — APPROVED** [review]

File: iothub/service/src/ServiceClient.cs (hunk 1 above)
Comment (abhipsaMisra) [reply iothub/service/src/ServiceClient.cs:277]: oh, breaking change for the amqp library then!

**Review by abhipsaMisra — APPROVED** [review]

File: iothub/service/src/ServiceClient.cs (hunk 1 above)
Comment (azabbasi) [reply iothub/service/src/ServiceClient.cs:277]: :D

**Pull request merged**